Improve Swift documentation (#11013)
DataDog/documentation: content/en/continuous_integration/setup_tests/swift.md

1. Add `dd-sdk-swift-testing` package to your project. It is located at [`https://github.com/DataDog/dd-sdk-swift-testing`][1].

>     ![](swift_images/swift_package.png)

Reviewer: This doesn't seem to work @nachoBonafonte ?

Author: It's standard markdown, I thought it would be supported

Author: Finally fixed

Reviewer: thanks

2. Copy and link your test targets with the resulting XCFramework.

    ![](swift_images/swift_link.png)

3. If you run UITests, also link the app running the tests with this library.

## Instrumenting your tests

### Configuring Datadog

To enable testing instrumentation, add the following environment variables to your test target. You must also select your main target in the `Expand variables based on:`  combo or `Target for Variable Expansion` if using test plans:

![](swift_images/swift_env.png)

For UITests, environment variables need to be set only in the test target, because the framework automatically injects these values to the application.

You should set all these variables in your test target:

`DD_TEST_RUNNER`

> : Enables or disables the instrumentation of tests. You should set this value to `$(DD_TEST_RUNNER)` so you can enable and disable test instrumentation with a environment variable<br/>

Reviewer: ```suggestion
: Enables or disables the instrumentation of tests. You should set this value to `$(DD_TEST_RUNNER)` so you can enable and disable test instrumentation with a environment variable defined outside of the test process (e.g. in the CI build).<br/>
```

**Default**: `(empty)`<br/>
**Recommended**: `$(DD_TEST_RUNNER)`<br/>
**Example**: `true`

`DATADOG_CLIENT_TOKEN`
: Your current Datadog client token. If you dont have one, you can create one [here][3]<br/>
**Default**: `(empty)`<br/>
**Examples**: `pub0zxxxyyyxxxyyxxxzzxxyyxxxyyy`

`DD_SERVICE`
: Name of the service or library under test.<br/>
**Default**: `(Repository name)`<br/>
**Example**: `my-marvelous-app`

`DD_ENV`
: Name of the environment where tests are being run. You should set this value to $(DD_ENV) so you can use an environment variable for setting it<br/>
**Default**: `(empty)`<br/>
**Recommended**: `$(DD_ENV)`<br/>
**Examples**: `ci`, `local`

`SRCROOT`
: This is the path to the project SRCROOT environment variable. Best option is using `$(SRCROOT)` for the value, so it uses Xcode internal variable<br/>
**Default**: `(empty)`<br/>
**Recommended**: `$(SRCROOT)`<br/>
**Example**: `/Users/ci/source/MyApp`

`DD_ENDPOINT`
: This is not needed if your company is using US endpoint<br/>
**Default**: `us`<br/>
**Examples**: `eu1`

### Configuring your CI service

> You must also set environment variables for your CI service. Note that Git information is automatically collected if you use a simulator to perform your MacOS, iOS, or tvOS tests. However, If you do testing on physical devices, you must also set the additional Git-related environment variables. You can find a list of the CI services and the environment variables at the end of this document, in the section [CI Services](#CI-Services)

Reviewer: ```suggestion
Git metadata and build information is automatically collected using CI provider environment variables, that must be forwarded to the test application (refer to the section [CI provider environment variables](#CI-provider-environment-variables) below for a full list).

When running tests in a simulator, full Git metadata is collected using the local `.git` folder. In this case, Git-related environment variables don't have to be forwarded.
```

### Running tests

After installation, you can run your tests as you normally do, for example using the `xcodebuild test` command. Tests, network requests, and application logs will be instrumented automatically. You can pass your environment variables when running your tests in the CI, e.g:

{{< code-block lang="bash" >}}
DD_TEST_RUNNER=1 DD_ENV=ci xcodebuild -project "MyProject.xcodeproj" -scheme "MyScheme" -destination "platform=macOS,arch=x86_64" test
{{< /code-block >}}

### UI tests

For UITests, both the test target and the application running from the UITests must link with the framework. Environment variables need to be set only in the test target, because the framework automatically injects these values to the application.

## Additional optional configuration

For the following configuration settings:
 - `Boolean` variables can use any of: `1`, `0`, `true`, `false`, `YES`, or `NO`
 - `String` list variables accept a list of elements separated by `,` or `;`

### Disabling auto-instrumentation

> The framework automatically captures the most information possible, but for some situations or tests this can be counter-productive. You can disable some of the auto-instrumentation for all the tests by setting environment variables:

Reviewer: ```suggestion
The framework enables auto-instrumentation of all supported libraries, but in some cases this might not be desired. You can disable auto-instrumentation of certain libraries by setting the following environment variables:
```

`DD_DISABLE_NETWORK_INSTRUMENTATION`
: Disables all network instrumentation (Boolean)

`DD_DISABLE_STDOUT_INSTRUMENTATION`
: Disables all `stdout` instrumentation (Boolean)

`DD_DISABLE_STDERR_INSTRUMENTATION`
: Disables all `stderr` instrumentation (Boolean)

`DD_DISABLE_SDKIOS_INTEGRATION`
: Disables integration with `dd-sdk-ios` logs and traces (Boolean)

`DD_DISABLE_CRASH_HANDLER`
: Disables crash handling and reporting. (Boolean)

> <div class="alert alert-warning"><strong>Important</strong>: If you disable crash reporting, crashing tests won't be reported to the backend and won't appear as failures. If you need to disable crash handling for any of your tests, run it as a separate target, so you don't disable it for the rest of the tests.</div>

Reviewer: ```suggestion
<div class="alert alert-warning"><strong>Important</strong>: If you disable crash reporting, tests that crash won't be reported at all, and won't appear as test failures. If you need to disable crash handling for any of your tests, run them as a separate target, so you don't disable it for the others.</div>
```

### Network auto-instrumentation

For Network auto-instrumentation, you can configure these additional settings:

`DD_DISABLE_HEADERS_INJECTION`
: Disables all injection of tracing headers (Boolean)

`DD_INSTRUMENTATION_EXTRA_HEADERS`
: Specific extra headers that you want to log (String List)

`DD_EXCLUDED_URLS`
: URLs that you don't want to log or inject headers into (String List)

`DD_ENABLE_RECORD_PAYLOAD`
: Enables reporting a subset (1024 bytes) of the payloads in requests and responses (Boolean)

`DD_MAX_PAYLOAD_SIZE`
: Sets the maximum size reported from the payload. Default `1024` (Integer)

You can also disable or enable specific auto-instrumentation in some of the tests from Swift or Objective-C by importing the module `DatadogSDKTesting` and using the class: `DDInstrumentationControl`.

> ### CI Services

Reviewer: ```suggestion
## CI provider environment variables
```
